src/terminal/sgr.zig

```diff
@@ -189,26 +189,39 @@ pub const Parser = struct {
                 .@"8_fg" = @enumFromInt(slice[0] - 30),
             },
 
-            38 => if (slice.len >= 5 and slice[1] == 2) {
-                self.idx += 4;
-
-                // In the 6-len form, ignore the 3rd param.
-                const rgb = slice[2..5];
-
-                // We use @truncate because the value should be 0 to 255. If
-                // it isn't, the behavior is undefined so we just... truncate it.
-                return Attribute{
-                    .direct_color_fg = .{
-                        .r = @truncate(rgb[0]),
-                        .g = @truncate(rgb[1]),
-                        .b = @truncate(rgb[2]),
-                    },
-                };
-            } else if (slice.len >= 3 and slice[1] == 5) {
-                self.idx += 2;
-                return Attribute{
-                    .@"256_fg" = @truncate(slice[2]),
-                };
+            38 => switch (slice[1]) {
```

Comment:
Do we need to check that `slice.len >= 2` or is that guaranteed to be true at this point?

Reply:
Ah, good catch thanks!